pkg/securelaunch/helpers.go

```diff
@@ -0,0 +1,270 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package securelaunch
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/u-root/u-root/pkg/cmdline"
+	"github.com/u-root/u-root/pkg/diskboot"
+	"github.com/u-root/u-root/pkg/find"
+	"github.com/u-root/u-root/pkg/mount"
+	"github.com/u-root/u-root/pkg/storage"
+)
+
+/* used to store all block devices returned from a call to storage.GetBlockStats */
+var storageBlkDevices []storage.BlockDev
+
+/*
+ * if kernel cmd line has uroot.uinitargs=-d, debug fn is enabled.
+ * kernel cmdline is checked in sluinit.
+ */
+var Debug = func(string, ...interface{}) {}
+
+/*
+ * modified ver of https://stackoverflow.com/questions/10510691/how-to-check-whether-a-file-or-directory-exists
+ * exists returns whether the given file or directory exists
+ * also returns if the path is a directory
+ */
+func exists(path string) (bool, bool, error) {
+	fileInfo, err := os.Stat(path)
+	if err == nil {
+		return true, fileInfo.IsDir(), nil
+	}
+	if os.IsNotExist(err) {
+		return false, false, nil
+	}
+	// file or dir def exists..
+	return true, fileInfo.IsDir(), err
+}
+
+/*
+ * data is a []byte slice
+ * dst is an already mounted file path.
+ * dst is an Absolute file path.
+ * defFileName is default dst file name, only used if user doesn't provide one.
+ * returns
+ * 1. target file path where file was written
+ * 2. error
+ */
+func WriteToFile(data []byte, dst, defFileName string) (string, error) {
+
+	// make sure dst is an absolute file path
+	if !filepath.IsAbs(dst) {
+		return "", fmt.Errorf("Error: dst =%s Not an absolute path ", dst)
+	}
+
+	// target is the full absolute path where []byte will be written to
+	target := dst
+
+	/* dst = /foo/bar/eventlog, check if /foo/bar/eventlog is a dir or file that exists..
+	     * if it exists as a file, this if loop is untouched.
+		 * if it exists as a dir, append defFileName to it
+	     * NOTE both dst = /foo/bar/ and dst=/foo/bar are considered dirs i.e. trailing "/" has no meaning..
+	*/
+	dstFound, is_dir, err := exists(dst)
```

Comment:
I feel like this would be more readable if it was inline, and it would eliminate the case below that can't happen:

```go
target := dst
dstInfo, err := os.Stat(dst)
if err != nil && !os.IsNotExist(err) {
  return err
} else if err == nil && dstInfo.IsDir() {
  target = filepath.Join(dst, defFileName)
}

if err := ioutil.WriteFile(target, ...)
```

Reply:
After ur above change it looks a lot more readable:
    `target := dst

   dstFound, is_dir, err := exists(dst)
    if is_dir {
        target = filepath.Join(dst, defFileName)
    }

    err = ioutil.WriteFile(target, data, 0644)
`

Reply:
That's still missing the error path

```go
target := dst

_, is_dir, err := exists(dst)
if err != nil {
  return err 
} else if is_dir {
target = filepath.Join(dst, defFileName)
}
```

Reply:
I feel like at that point it's equivalent to what I suggested further up, except you can delete `exists` altogether

Reply:
Hmm, exists is also called by other functions such as WriteSrcFiletoDst() ...i would need to make it inline in multiple places...not sure if i like that..

Reply:
If you look at my comment on WriteSrcFileToDst, exists is also unnecessary there

Reply:
and i think that's the two callers i can see?